Source/DafnyCore/Compilers/GoLang/Compiler-go.cs

```diff
@@ -1281,37 +1306,32 @@ protected class ClassWriter : IClassWriter {
       foreach (var tp in typeParams) {
         if (useAllTypeArgs || NeedsTypeDescriptor(tp)) {
           var name = FormatRTDName(tp.GetCompileName(Options));
-
-          if (wr != null) {
-            wr.WriteLine($"{name} {DafnyTypeDescriptor}");
-          }
-
-          if (wInit != null) {
-            wInit.WriteLine("_this.{0} = {0}", name);
-          }
-
-          if (wParams != null) {
-            wParams.Write($"{sep}{name} {DafnyTypeDescriptor}");
-            sep = ", ";
-          }
-
+          wr?.WriteLine($"{name} {DafnyTypeDescriptor}");
+          wInit?.WriteLine("_this.{0} = {0}", name);
+          wParams?.Write($"{sep}{name} {DafnyTypeDescriptor}");
+          sep = ", ";
           count++;
         }
       }
       return count;
     }
 
-    void WriteRuntimeTypeDescriptorsFormals(List<TypeParameter> typeParams, bool useAllTypeArgs, ConcreteSyntaxTree wr) {
```

Comment:
Consider returning `ICanRender` instead of taking and mutating `ConcreteSyntaxTree`, so more work is done by the callee.